### PR TITLE
Add `Context` overloads to `AssertEx.Builder` chains

### PR DESCRIPTION
**Pull request by michael-sterling**
Also, remove dependencies on `Faithlife.Uitlity` and `System.Collections.Immutable` as the amount of code-duplication they relieve us of is far smaller than the cost they impose in version-conflicts.

### CODE REVIEW

File: src/Faithlife.Testing/AssertEx.cs [hunk 1]
```diff
@@ -73,33 +72,8 @@ public static T Select<T>(Expression<Func<T?>> valueExpression)
...
-					}
-					catch (Exception ex)
-					{
-						return (object) ex;
-					}
-				}, LazyThreadSafetyMode.ExecutionAndPublication)));
-			}));
-		}
+		public static IDisposable Context(object context) => Context(GetContextFromObject(context));
+		public static IDisposable Context(params Expression<Func<object>>[] contextExpressions) => Context(GetContextFromExpressions(contextExpressions));
```
Comment (josephdangerstewart) [inline src/Faithlife.Testing/AssertEx.cs:76]: Nit: Can we move these down with the other `Context` overloads?

Comment (josephdangerstewart) [reply src/Faithlife.Testing/AssertEx.cs:76]: Oh wait nevermind, I missed the `static` here

File: ReleaseNotes.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Release Notes
 
+## 1.3.0
+
+* Minor: Added `Context` overloads to `AssertEx.Builder` chains
+* Patch: Remove dependencies on `Faithlife.Uitlity` and `System.Collections.Immutable`.
```
Comment (RobertBolender) [inline ReleaseNotes.md:6]:
```suggestion
* Patch: Remove dependencies on `Faithlife.Utility` and `System.Collections.Immutable`.
```

File: src/Faithlife.Testing.RabbitMq/MessagePublishedAwaiter.cs [hunk 3]
```diff
@@ -36,7 +35,7 @@ public MessagePublishedAwaiter(string serverName, string exchangeName, string ro
 
 			m_model = m_connection.CreateModel();
 
-			var queueName = $"{exchangeName}_{routingKeyName}_awaiter_{Environment.MachineName}_{Guid.NewGuid().ToLowerNoDashString()}";
+			var queueName = $"{exchangeName}_{routingKeyName}_awaiter_{Environment.MachineName}_{Guid.NewGuid():N}";
```
Comment (josephdangerstewart) [inline src/Faithlife.Testing.RabbitMq/MessagePublishedAwaiter.cs:38]: > `{Guid.NewGuid():N}`

TIL: String formatting in interpolations

**Review by josephdangerstewart — APPROVED** [review]

**Pull request merged**